semver.js

```diff
@@ -704,6 +704,40 @@ Comparator.prototype.test = function(version) {
   return cmp(version, this.operator, this.semver, this.loose);
 };
 
+Comparator.prototype.intersects = function(comp, loose, platform) {
+  var rangeTmp;
+
+  if (this.operator === '') {
+    rangeTmp = new Range(comp.value, loose, platform);
+    return satisfies(this.value, rangeTmp, loose, platform);
+  } else if (comp.operator === '') {
+    rangeTmp = new Range(this.value, loose, platform);
+    return satisfies(comp.semver, rangeTmp, loose, platform);
+  }
+  // Same direction increasing
+  return ((this.operator === '>=' || this.operator === '>') && (comp.operator === '>=' || comp.operator === '>')) ||
```

Comment:
Sure - maybe just separate variables would help. Here's a subset:
```js
var sameDirectionIncreasing = (this.operator === '>=' || this.operator === '>') && (comp.operator === '>=' || comp.operator === '>');
var sameDirectionDecreasing = (this.operator === '<=' || this.operator === '<') && (comp.operator === '<=' || comp.operator === '<');
var sameSemver = this.semver.raw === comp.semver.raw;
var differentDirectionsInclusive = (this.operator === '>=' || this.operator === '<=') && (comp.operator === '>=' || comp.operator === '<=');
return sameDirectionIncreasing || sameDirectionDecreasing || (sameSemver && differentDirectionsInclusive);
```

something like that? the long chain of boolean operations makes it hard for me to read.